pkg/osvscanner/osvscanner_internal_test.go

```diff
@@ -59,17 +47,14 @@ func Test_filterResults(t *testing.T) {
 				DefaultConfig: config.Config{},
 				ConfigMap:     make(map[string]config.Config),
 			}
-			got := tt.testCase.input
+
+			got := testutility.LoadJSONFixture[models.VulnerabilityResults](t, filepath.Join(tt.path, "input.json"))
 			filtered := filterResults(r, &got, &configManager, false)
-			if diff := cmp.Diff(tt.testCase.want, got); diff != "" {
-				out := filepath.Join(tt.testCase.path, "out.json")
-				t.Errorf("filterResults() returned an unexpected results (-want, +got):\n%s\n"+
-					"Full json output written to %s", diff, out)
-				//nolint:staticcheck
-				testutility.CreateJSONFixture(t, out, got)
-			}
-			if filtered != tt.testCase.numFiltered {
-				t.Errorf("filterResults() = %v, want %v", filtered, tt.testCase.numFiltered)
+
+			testutility.NewSnapshot().MatchJSON(t, got)
+
+			if filtered != tt.want {
```

Comment:
Will this get called when MatchJSON does not match? If it doesn't, we might want to move this up above line 54, as this could be more useful information at a glance compared to the full snapshot diff.

Reply:
This is just the count of items that got filtered, so I think it would be worse to have it fail first because it wouldn't tell you _which_ items were unexpected missing or present.

Reply:
I think because we use Errorf here it will flow through to the MatchJSON and we'll get both outputs. I'm just not sure if snapshot uses Fatalf or Errorf when it fails to match.

Reply:
ah yeah, `go-snaps` uses `t.Errorf` too:

```
--- FAIL: Test_filterResults (0.00s)
    --- FAIL: Test_filterResults/filter_partially (0.00s)
        osvscanner_internal_test.go:54:
            - Snapshot - 0
            + Received + 4

            @@ -547,5 +547,9 @@

                  }
                ],
                "experimental_config": {
            +     "licenses": {
            +       "summary": false,
            +       "allowlist": null
            +     }
                }
              }

            at ../__snapshots__/osvscanner_internal_test.snap:1476

        osvscanner_internal_test.go:57: filterResults() = 10, want 17
FAIL
coverage: 7.0% of statements in ./...

Snapshot Summary

✓ 2 snapshots passed
✕ 1 snapshot failed
```